tests/unit/models/027-shelf.js

```diff
@@ -0,0 +1,118 @@
+const CONFIG = require('config')
+const __ = CONFIG.universalPath
+const _ = __.require('builders', 'utils')
+const Shelf = __.require('models', 'shelf')
+require('should')
+
+const someUserId = '1234567890a1234567890b1234567890'
+const create = Shelf.create.bind(null)
+const update = Shelf.updateAttributes.bind(null)
+
+const faker = require('faker')
+const fakeName = faker.random.words(4)
+const fakeDesc = faker.random.words(15)
+
+const validShelf = {
+  owner: someUserId,
+  description: fakeDesc,
+  listing: 'private',
+  name: fakeName
+}
+
+const extendShelf = data => Object.assign({}, validShelf, data)
+
+describe('shelf model', () => {
+  describe('create', () => {
+    it('should return an object', done => {
+      const shelf = create(validShelf)
+      shelf.should.be.an.Object()
+      shelf.name.should.equal(fakeName)
+      shelf.description.should.equal(fakeDesc)
+      shelf.owner.should.equal(someUserId)
+      shelf.listing.should.equal('private')
+      shelf.created.should.be.a.Number()
+      done()
+    })
+
+    it('should throw when passed an invalid attributes', done => {
+      const shelf = extendShelf({ authors: 'Abel Paz' })
+      const creator = () => Shelf.create(shelf)
+      creator.should.throw()
+      done()
+    })
+
+    describe('mandatory attributes', () => {
+      it('should throw on missing name', done => {
+        const invalidShelf = _.cloneDeep(validShelf)
+        delete invalidShelf.owner
+        const creator = () => Shelf.create(invalidShelf)
+        creator.should.throw()
+        done()
+      })
+
+      it('should throw on missing name', done => {
+        const invalidShelf = _.cloneDeep(validShelf)
+        delete invalidShelf.name
+        const creator = () => Shelf.create(invalidShelf)
+        creator.should.throw()
+        done()
+      })
+    })
+
+    describe('listing', () => {
+      it('should use a default listing value', done => {
+        const shelf = create(extendShelf({ listing: null }))
+        shelf.listing.should.equal('private')
+        done()
+      })
+
+      it('should override a bad listing with default value', done => {
```

Comment:
that's sounds like failing silently: as a client developer, I prefer getting a 400 error due to a typo and thus be able to kickly fix it, rather than having the server silently do something else than what I intended

Reply:
as you know im a "convention over config" type, id rather have a server which fill my request than saying "you have a typo here"

Reply:
vegan customer: "I would like a vegna sandwich please"
server: "yes sir, here is your jambon-beurre"

how is that a convention? ^^

Reply:
its a POST not a GET so its more "- hey here is the box, can you store it in a moqsifjoi place please ?" in the servers head: "This person is having trouble understanding how its works here, lets put it in "private place" and tell them I did so"

Reply:
~and tell them I did so~ and let them discover later that something somewhere in their code or ours went differently than expected, without having any pointer at where that could be

it seems to me that the sensible defaults (to go on the convention over configuration approach) would be to accept an empty listing value and thus default to "private", not to accept invalid values and silently overwrite them

Reply:
alright

Reply:
that would be the consistent thing to do indeed ^^